src/AbstractSQLCompiler.ts

```diff
@@ -746,6 +747,66 @@ export function compileRule(
 	return AbstractSQLRules2SQL(abstractSQL, engine, noBinds);
 }
 
+// Define sorting weights by data types
+// TODO: Add support for all possible data types
+// TODO: Decide if we want to try and always start with 'id', 'created at', 'modified at'
+const sortWeightByType: Record<string, number> = {
+	'Big Integer': 1,
+	'Big Serial': 1,
+	'Date Time': 1,
+	Integer: 2,
+	Serial: 2,
+	'Short Text': 3,
+	File: 4,
+	Text: 4,
+	// eslint-disable-next-line id-denylist
+	Boolean: 5,
+} as const;
```

Comment:
We could potentially also store sorting weights as a property of the types themselves in `sbvr-types`.